packages/foam-vscode/jest.config.js

```diff
@@ -152,7 +152,7 @@ module.exports = {
   // ],
 
   // The regexp pattern or array of patterns that Jest uses to detect test files
-  // testRegex: [],
+  testRegex: ['\\.test\\.ts$'],
```

Comment:
This is overridden in every `runCLI` invocation but it's here as the default for vscode-jest (since there is no extension specific setting for this AFAICS). We only want unit tests in the test explorer (sidebar), since spec tests require the entire extension host to be launched before.

Reply:
This is a very important point, above in the `vscode-jest-tests` I was about to ask exactly whether it was supporting both unit and integration tests.
I think we need to either highlight this connection, or better yet make the `vscode-jest-tests` self-explanatory:
- we could add this comment somewhere in the code, to "link" the `vscode-jest-tests` to this configuration, or
- we could have this setting as part of the `vscode-jest-tests` configuration, so that it doesn't have a dependency on this external configuration option
> since there is no extension specific setting for this AFAICS

Could you use the `args` param? Or maybe that's only used at runtime so the extension would still "flag" e2e tests as runnable in VS Code?

What do you think?

Reply:
> `vscode-jest-tests`

Notice that the extension I'm using is `vscode-jest`, this one: https://github.com/jest-community/vscode-jest

>  I was about to ask exactly whether it was supporting both unit and integration tests.

The sidebar explorer offers to run each test separately. In principle, we could write a script that launches the entire extension host just to run some e2e test and register this script as the `vscode-jest-tests` launcher, but it doesn't seem very useful because of the high overhead of creating the environment. Moreover, the launcher would be of type `node` (I don't think `vscode-jest` is able to run other types), so really unsupported as a way of debugging a test running inside the host.

The use case of quickly running some unit test from the sidebar is convenient and simple enough to support, so I'm focusing on that.

> we could add this comment somewhere in the code, to "link" the vscode-jest-tests to this configuration

I've added the comment to `jest.config.js`.

> we could have this setting as part of the vscode-jest-tests configuration

This was the first thing I tried, but unfortunately I found no such setting.

> Could you use the args param? 

No, the launcher is used to run a test but not to build the sidebar tree.